### PR TITLE
feat(ecc-utils-design): adds collapsible group fields to the form

### LINKED ISSUE #177: feat: collapsable view to form component
## Description
Some of the forms requires being grouped inside a subsection which needs to be an accordion which will improve the user experience. This will be very useful for the forms which are very long as they can collapsed down. 

## Possible solution(s)

Maybe we can have extra `collapsible` based on which we can render the template, if certain fields are grouped inside `collapsible` then we can render as we doing right now but inside an accordion and otherwise we can do the same as we are doing currently.

### PR DESCRIPTION
**Pull request by SalihuDickson**
## Description

This PR adds the feature of collapsible form groups to the form component. When the type of a field is set to `group` it groups fields listed as the children together in the form. And when the `groupOption.collapsible` is set to true for that field the field will be rendered as an accordion allowing the field to be expanded and collapsed to reveal or hide its children. 

Fixes #177 

## Checklist

- [ ] My code follows the [contributing guidelines][contributing] of this project.
- [ ] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

### CODE REVIEW

**Comment by JaeAeich** [comment]
@SalihuDickson can you please Docs for it as well :slightly_smiling_face:

**Comment by SalihuDickson** [comment]
@JaeAeich, sure I'll add examples in the docs (I've already created an issue to make the docs more detailed). But I'm  a little busy right now with other things so In the mean time, you can just use it sort of like the field of type array, set the field type to `group` and then add the fields that are supposed to be in the group as its children. To make it collapsible, simply the `groupOptions.collapsible` to `true`.

**Comment by JaeAeich** [comment]
LGTM, maybe @git-anurag-hub can take a look as well.

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 1]
```diff
@@ -218,6 +226,7 @@ export default class Form extends LitElement {
           }
 
           this.requestUpdate();
+          console.log(this.form);
```
Comment (JaeAeich) [inline packages/ecc-utils-design/src/components/form/form.ts:229]: ?

**Comment by SalihuDickson** [comment]
> @SalihuDickson can you resolve the conflicts with the `main` for merge! Thanks

done!

**Pull request merged**